wit/types.wit

```diff
@@ -16,29 +25,50 @@ interface messaging-types {
     /// The interface doesn't highlight this difference in the type itself as that's uniquely a consumer issue.
     type channel = string;
   
-    /// Configuration includes a required list of channels the guest is subscribing to, and an optional list of extensions key-value pairs 
-    /// (e.g., partitions/offsets to read from in Kafka/EventHubs, QoS etc.).
+    /// Configuration includes a required list of channels the guest is subscribing to, and an
+    /// optional list of extensions key-value pairs (e.g., partitions/offsets to read from in 
+    /// Kafka/EventHubs, QoS etc.).
     record guest-configuration {
         channels: list<channel>,
         extensions: option<list<tuple<string, string>>>
     }
   
-    /// Format specification for messages 
-    ///  - more info: https://github.com/clemensv/spec/blob/registry-extensions/registry/spec.md#message-formats
-    ///  - message metadata can further decorate w/ things like format version, and so on.
-    enum format-spec {
-        cloudevents,
-        http,
-        amqp,
-        mqtt,
-        kafka,
-        raw
-    }
-  
-    /// A message with a binary payload, a format specification, and decorative metadata.
-    record message {
-        data: list<u8>,
-        format: format-spec,
-        metadata: option<list<tuple<string, string>>>
+    /// A message with a binary payload and additional information
+    resource message {
+        constructor(topic: channel, data: list<u8>, content-type: option<string>, metadata: option<list<tuple<string, string>>>);
+        /// The topic/subject/channel this message was received or should be sent on
+        topic: func() -> channel;
```

Comment:
Looking through the types, naming of parameters, and comments, I'm getting a little turned around with the difference between a topic/subject/channel. Taking a look at pubsub messaging systems, I know that Apache Kafka, Google Cloud Pub/Sub, and Confluent use "topic", NATS uses "subject" but also uses that interchangeably with "topic", and then Redis uses channels. These are just a few options and I can totally see where the naming here came from.

This might be a bikeshed, but what are your thoughts on just calling this a "topic" all around the interface, and then clarifying on the type itself, e.g.
```
    /// Topics, aka subjects, channels, groups in pubsub systems
    /// ...
    type topic = string,
```

Since it's a strawman I'm fine considering any terminology here, just suggesting to keep it consistent

Reply:
I think topic would be a lot clearer personally. Any thoughts @danbugs since I think you might have been the one who picked "channel"

Reply:
Iirc, I just chose `channel` to avoid picking a specific nomenclature 👍

Reply:
I've kept it channel for now. Let me know if there are any strong opinions to the contrary

Reply:
IMHO the term `channel` implies a "transport" from one end to the other. The alternatives `topic` and `subject` are neutral to that aspect. But they have the advantage that they imply to "address" the message source. The messaging (at least in the IIoT space) is organized around an "asset" not the "channel" the messages are sent through. So, I prefer `topic` or `subject`.

Reply:
Those sound like great reasons to me. I've updated this to be `topic`